pandas/core/construction.py

```diff
@@ -119,7 +119,7 @@ def array(
         :class:`datetime.timedelta`    :class:`pandas.arrays.TimedeltaArray`
         :class:`int`                   :class:`pandas.arrays.IntegerArray`
         :class:`float`                 :class:`pandas.arrays.FloatingArray`
-        :class:`str`                   :class:`pandas.arrays.StringArray`
+        :class:`str`                   :class:`pandas.arrays.PythonStringArray`
```

Comment:
there is overlap here with the dtype change. could add the docstring to the baseclass and no need to reference the two array types.